test/stdlib/CodableTests.swift

```diff
@@ -1100,6 +1100,152 @@ class TestCodable : TestCodableSuper {
             expectRoundTripEqualityThroughPlist(for: UUIDCodingWrapper(uuid), lineNumber: testLine)
         }
     }
+
+    // MARK: - DecodingError
+    func expectErrorDescription(
+        _ expectedErrorDescription: String,
+        fromDecodingError error: DecodingError,
+        lineNumber: UInt = #line
+    ) {
+        expectEqual(String(describing: error), expectedErrorDescription, "Unexpectedly wrong error: \(error)", line: lineNumber)
+    }
+
+    func test_decodingError_typeMismatch_nilUnderlyingError() {
+        expectErrorDescription(
+            #"""
+            This is where the debug description goes
+            Path: [0]/address/city/birds/[1]/name
+            """#,
+            fromDecodingError: DecodingError.typeMismatch(
+                String.self,
+                DecodingError.Context(
+                    codingPath: [0, "address", "city", "birds", 1, "name"] as [GenericCodingKey],
+                    debugDescription: "This is where the debug description goes"
+                )
+            )
+        )
+    }
+
+    func test_decodingError_typeMismatch_nonNilUnderlyingError() {
+        expectErrorDescription(
+            #"""
+            Some debug description
+            Underlying error: GenericError(name: "some generic error goes here")
+            Path: [0]/address/[1]/street
+            """#,
+            fromDecodingError: DecodingError.typeMismatch(
+                String.self,
+                DecodingError.Context(
+                    codingPath: [0, "address", 1, "street"] as [GenericCodingKey],
+                    debugDescription: "Some debug description",
+                    underlyingError: GenericError(name: "some generic error goes here")
+                )
+            )
+        )
+    }
+
+    func test_decodingError_valueNotFound_nilUnderlyingError() {
+        expectErrorDescription(
+            #"""
+            Description for debugging purposes
+            Path: [0]/firstName
+            """#,
+            fromDecodingError: DecodingError.valueNotFound(
+                String.self,
+                DecodingError.Context(
+                    codingPath: [0, "firstName"] as [GenericCodingKey],
+                    debugDescription: "Description for debugging purposes"
+                )
+            )
+        )
+    }
+
+    func test_decodingError_valueNotFound_nonNilUnderlyingError() {
+        expectErrorDescription(
+            #"""
+            Here is the debug description for value-not-found
+            Underlying error: GenericError(name: "these aren\'t the droids you\'re looking for")
```

Comment:
I'm not sure where the `\'` escape sequences are coming from 🤔

Reply:
Hmm, it looks like it might be from `String(reflecting:)` vs `String(describing:)`. I don't think we want it, but it might be unavoidable if we want to use `String(debugDescription:)`, which seems appropriate given that we're inside a conformance to `CustomDebugStringConvertible`.

<img width="443" alt="image" src="https://github.com/user-attachments/assets/2171d8c7-3228-49c5-87dc-bb83e1d19882" />